plugins/inputs/pgbouncer/pgbouncer.go

```diff
@@ -20,40 +21,121 @@ var sampleConfig string
 
 type PgBouncer struct {
 	postgresql.Service
+	ShowCommands []string `toml:"show_commands"`
 }
 
 var ignoredColumns = map[string]bool{"user": true, "database": true, "pool_mode": true,
 	"avg_req": true, "avg_recv": true, "avg_sent": true, "avg_query": true,
+	"force_user": true, "host": true, "port": true, "name": true,
 }
 
 func (*PgBouncer) SampleConfig() string {
 	return sampleConfig
 }
 
 func (p *PgBouncer) Gather(acc telegraf.Accumulator) error {
-	var (
-		err     error
-		query   string
-		columns []string
-	)
+	if len(p.ShowCommands) == 0 {
+		if err := p.showStats(acc); err != nil {
+			return err
+		}
+
+		if err := p.showPools(acc); err != nil {
+			return err
+		}
+	} else {
+		for _, cmd := range p.ShowCommands {
+			switch {
+			case cmd == "stats":
+				if err := p.showStats(acc); err != nil {
+					return err
+				}
+			case cmd == "pools":
+				if err := p.showPools(acc); err != nil {
+					return err
+				}
+			case cmd == "lists":
+				if err := p.showLists(acc); err != nil {
+					return err
+				}
+			case cmd == "databases":
+				if err := p.showDatabase(acc); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+type scanner interface {
+	Scan(dest ...interface{}) error
+}
+
+func (p *PgBouncer) accRow(row scanner, columns []string) (map[string]string, map[string]*interface{}, error) {
+	var dbname bytes.Buffer
+
+	// this is where we'll store the column name with its *interface{}
+	columnMap := make(map[string]*interface{})
+	for _, column := range columns {
+		columnMap[column] = new(interface{})
+	}
+
+	columnVars := make([]interface{}, 0, len(columnMap))
+	// populate the array of interface{} with the pointers in the right order
+	for i := 0; i < len(columnMap); i++ {
+		columnVars = append(columnVars, columnMap[columns[i]])
+	}
+
+	// deconstruct array of variables and send to Scan
+	err := row.Scan(columnVars...)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't copy the data: %w", err)
+	}
+	if columnMap["database"] != nil {
+		// extract the database name from the column map
+		name, ok := (*columnMap["database"]).(string)
+		if !ok {
+			return nil, nil, fmt.Errorf("database not a string, but %T", *columnMap["database"])
+		}
+		_, err := dbname.WriteString(name)
+		if err != nil {
+			return nil, nil, fmt.Errorf("writing database name failed: %w", err)
+		}
+	} else {
+		_, err := dbname.WriteString("postgres")
+		if err != nil {
+			return nil, nil, fmt.Errorf("writing database name failed: %w", err)
+		}
```

Comment:
Here both prefixes are the same which defeats the purpose... Assume a user reports an error and the error message says `writing database name failed: something odd happened` which of the `WriteString()`  calls above triggered it? You cannot tell, so I suggest
```suggestion
		if err != nil {
			return nil, nil, fmt.Errorf("writing database name failed: %w", err)
		}
	} else {
		_, err := dbname.WriteString("postgres")
		if err != nil {
			return nil, nil, fmt.Errorf("writing 'postgres' failed: %w", err)
		}
```
as you now know for sure where the error happened!